src/logger.ts

```diff
@@ -0,0 +1,93 @@
+import * as os from 'os';
+import * as path from 'path';
+import * as vscode from 'vscode';
+import * as winston from 'winston';
+import * as TransportStream from 'winston-transport';
+
+class OutputChannelTransport extends TransportStream {
```

Comment:
If we took this without modification from somewhere, can we link to the source (for credit and also future maintenability)?

Reply:
I think this is based on Winston documentation about how to define custom transports: https://github.com/winstonjs/winston/blob/2625f60c5c85b8c4926c65e98a591f8b42e0db9a/README.md#adding-custom-transports.  The extra bit is that here the class has an attribute

Reply:
Okay, I thought it had come from another example for some reason!